integrations/tests/test_shuffle.py

```diff
@@ -0,0 +1,193 @@
+            patch('requests.post', return_value=requests.Response):
+        shuffle.main(sys_args_template)
+
+
+def test_process_args_alert_file_exit():
+    """Test that process_args function exits when alert file is not found."""
+    with patch("builtins.open") as open_mock, \
+            pytest.raises(SystemExit) as pytest_wrapped_e:
+        open_mock.side_effect = FileNotFoundError
+        shuffle.process_args(sys_args_template)
+    assert pytest_wrapped_e.value.code == 3
+
+
+def test_process_args():
+    """Test the correct execution of the process_args function."""
+    with patch("builtins.open", mock_open()) as alert_file, \
+            patch('json.load', return_value=alert_template), \
+            patch('shuffle.send_msg') as send_msg, \
+            patch('shuffle.generate_msg', return_value=msg_template) as generate_msg, \
+            patch('requests.post', return_value=requests.Response):
+        shuffle.process_args(sys_args_template)
+        alert_file.assert_called_once_with(sys_args_template[1])
+        generate_msg.assert_called_once_with(alert_template)
+        send_msg.assert_called_once_with(msg_template, sys_args_template[3])
+
+
+def test_process_args_json_exit(tmpdir):
+    """Test that the process_args function exits when json exception is raised.
+
+    Parameters
+    ----------
+    tmpdir: py.path.local
+        Path to a temporary directory generated for the tests logs.
+    """
+    log_file = tmpdir.join('test.log')
+    with patch('shuffle.debug_enabled', return_value=True), \
+            patch('shuffle.LOG_FILE', str(log_file)), \
+            patch("builtins.open", mock_open()), \
+            patch('json.load') as json_load, \
+            pytest.raises(SystemExit) as pytest_wrapped_e:
+        json_load.side_effect = json.decoder.JSONDecodeError("Expecting value", "", 0)
+        shuffle.process_args(sys_args_template)
+    assert pytest_wrapped_e.value.code == 4
+
+
+def test_process_args_not_sending_message():
+    """Test that the send_msg function is not executed due to empty message after generate_msg."""
+    with patch("builtins.open", mock_open()), \
+            patch('json.load', return_value=alert_template), \
+            patch('shuffle.send_msg') as send_msg, \
+            patch('shuffle.generate_msg', return_value=''):
+        shuffle.process_args(sys_args_template)
+        send_msg.assert_not_called()
+
+
+def test_debug(tmpdir):
+    """Test the correct execution of the debug function, writing the expected log when debug mode enabled.
+
+    Parameters
+    ----------
+    tmpdir: py.path.local
+        Path to a temporary directory generated for the tests logs.
+    """
+    log_file = tmpdir.join('test.log')
+    with patch('shuffle.debug_enabled', return_value=True), \
+            patch('shuffle.LOG_FILE', str(log_file)):
+        shuffle.debug(msg_template)
+
+    assert log_file.read() == f"{shuffle.now}: {msg_template}\n"
```

Comment:
Never make use of temporary files in a unit test unless it is absolutely necessary. You can achieve the same thing by making that the open file returns a mocked object and checking `your_mock_object.write.assert_called_with(...)` instead of reading from a real log file. For example, you could use `MagickMock`.